pkg/certificateprovider/azurekeyvault/auth.go

```diff
@@ -64,13 +62,7 @@ func getAuthorizerForWorkloadIdentity(ctx context.Context, tenantID, clientID, r
 		return nil, fmt.Errorf("failed to acquire token: %w", err)
 	}
 
-	token := adal.Token{
-		AccessToken: result.AccessToken,
-		Resource:    resource,
-		Type:        tokenTypeBearer,
-	}
-	token.ExpiresOn, err = parseExpiresOn(result.ExpiresOn.UTC().Local().Format(expiresOnDateFormat))
-	if err != nil {
+	if _, err = parseExpiresOn(result.ExpiresOn.UTC().Local().Format(expiresOnDateFormat)); err != nil {
```

Comment:
is it easy to add unit test for " parseExpiresOn(result.ExpiresOn.UTC().Local().Format(expiresOnDateFormat))"

Reply:
The above line will invoke `/microsoft-authentication-library-for-go` library to make remote request to fetch AAD token which cannot be mocked, therefore this method is not unit tested yet.